plugins/wasi_nn/piper.cpp

```diff
@@ -18,64 +20,71 @@
 #include <piper.hpp>
 #include <string>
 #include <string_view>
+#include <tuple>
 #include <vector>
 #endif
 
 namespace WasmEdge::Host::WASINN::Piper {
 #ifdef WASMEDGE_PLUGIN_WASI_NN_BACKEND_PIPER
 
-template <typename T, typename U = T>
-Expect<WASINN::ErrNo> getOptionalOption(simdjson::dom::element &Doc,
-                                        std::string_view Key,
-                                        std::optional<T> &Result) {
-  if (Doc.at_key(Key).error() == simdjson::SUCCESS) {
-    auto Value = U{};
-    auto Err = Doc[Key].get<U>().get(Value);
-    if (Err) {
-      spdlog::error(
-          "[WASI-NN] Piper backend: Unable to retrieve the {} option."sv, Key);
-      return WASINN::ErrNo::InvalidArgument;
+template <typename T>
+std::tuple<WASINN::ErrNo, bool> getOption(simdjson::dom::object &Object,
+                                          std::string_view Key, T &Result) {
+  if (auto Error = Object[Key].get(Result)) {
+    if (Error == simdjson::error_code::NO_SUCH_FIELD) {
+      return {WASINN::ErrNo::Success, false};
     }
-    Result = Value;
+    spdlog::error(
+        "[WASI-NN] Piper backend: Unable to retrieve the \"{}\" option: {}"sv,
+        Key, simdjson::error_message(Error));
+    return {WASINN::ErrNo::InvalidArgument, false};
   }
-  return WASINN::ErrNo::Success;
+  return {WASINN::ErrNo::Success, true};
 }
 
-template <typename T>
-Expect<WASINN::ErrNo> getOption(simdjson::dom::element &Doc,
-                                std::string_view Key, T &Result) {
-  if (Doc.at_key(Key).error() == simdjson::SUCCESS) {
-    auto Err = Doc[Key].get<T>().get(Result);
-    if (Err) {
-      spdlog::error(
-          "[WASI-NN] Piper backend: Unable to retrieve the {} option."sv, Key);
-      return WASINN::ErrNo::InvalidArgument;
-    }
+template <typename T, typename U = T>
+WASINN::ErrNo getOptionalOption(simdjson::dom::object &Object,
+                                std::string_view Key,
+                                std::optional<T> &Result) {
+  auto Value = U{};
+  auto [Err, HasValue] = getOption(Object, Key, Value);
+  if (HasValue) {
+    Result = Value;
   }
-  return WASINN::ErrNo::Success;
+  return Err;
+}
+
+bool fileExists(const std::string &Filename) {
+  auto Stream = std::ifstream{Filename, std::ios::binary};
```

Comment:
I have a question. What if the `Filename` is a utf8string? Can it be handled in this way?

Reply:
Do you mean if `Filename` is UTF-8 encoded? I guess it depends on whether the native filesystem encoding is also UTF-8 :(

Reply:
How about using `std::filesystem::u8path` before passing to `std::ifstream`, which will ensure the u8string is handled correctly by the filesystem?

Reply:
Just like we did in the filesystem related tests. Ref: https://github.com/WasmEdge/WasmEdge/blob/9e9974411f444f588ca0cdd3550349ccabda4cf2/test/api/APIAOTCoreTest.cpp#L59-L61

Reply:
done in https://github.com/WasmEdge/WasmEdge/pull/3499/commits/0d13802d0a1e177103d95bfa33340e3c332206b6